### PR TITLE
Verified Id Decoder/Encoder

### PR DESCRIPTION
**Pull request by symorton**
- Add encode and decode methods on the VerifiedIdClient.
- Developers need a way to store raw VerifiedId data model within their apps, and encode and decode give them the ability to do that without needing to know the specific concrete class of the Verified Id object.
- Added tests.

### CODE REVIEW

File: WalletLibrary/WalletLibrary/VerifiedIdClientBuilder.swift [hunk 1]
```diff
@@ -26,7 +26,9 @@ public class VerifiedIdClientBuilder {
         let _ = VCServices.VerifiableCredentialSDK.initialize()
         
         let configuration = LibraryConfiguration(logger: logger,
-                                                 mapper: Mapper())
+                                                 mapper: Mapper(),
+                                                 verifiedIdDecoder: VerifiedIdDecoder(),
```
Comment (nithyaganeshng) [inline WalletLibrary/WalletLibrary/VerifiedIdClientBuilder.swift:32]: Is it instantiated both here and Configuration? Is the one in configuration like default value?

File: WalletLibrary/WalletLibraryTests/Mocks/VerifiedId/MockVerifiedId.swift
Comment (nithyaganeshng) [inline WalletLibrary/WalletLibraryTests/Mocks/VerifiedId/MockVerifiedId.swift:1]: Is only the primitive types enough for testing? Do we need map/list just to test special cases?

File: WalletLibrary/WalletLibraryTests/VerifiedIdClientTests.swift [hunk 2]
```diff
@@ -21,7 +23,8 @@ class VerifiedIdClientTests: XCTestCase {
         let mockInput = MockInput(mockData: "")
         
         let handlerFactory = RequestHandlerFactory(requestHandlers: [])
-        let configuration = LibraryConfiguration(logger: WalletLibraryLogger(), mapper: Mapper())
+        let configuration = LibraryConfiguration(logger: WalletLibraryLogger(),
+                                                 mapper: Mapper())
```
Comment (nithyaganeshng) [inline WalletLibrary/WalletLibraryTests/VerifiedIdClientTests.swift:27]: nit: Is this indentation intentional?

File: WalletLibrary/WalletLibrary/VerifiedIdClientBuilder.swift (hunk 1 above)
Comment (symorton) [reply WalletLibrary/WalletLibrary/VerifiedIdClientBuilder.swift:32]: Yea I just put it in default values for testing but theoretically, in the future we can inject the logger or something into the decoder and encoder here.

File: WalletLibrary/WalletLibraryTests/Mocks/VerifiedId/MockVerifiedId.swift
Comment (symorton) [reply WalletLibrary/WalletLibraryTests/Mocks/VerifiedId/MockVerifiedId.swift:1]: yes this should be enough, this mock is for when we don't care what the VerifiedId object is.

**Review by nithyaganeshng — APPROVED** [review]

**Pull request merged**